src/components/Tool/FixSendNewsletter/FixSendNewsletter.tsx

```diff
@@ -143,110 +110,122 @@ const FixSendNewsletter: React.FC<Props> = ({
+        <Grid item xs={12}>
+          <Typography variant="h4">{t('Fix Send Newsletter')}</Typography>
+          <Divider className={classes.divider} />
+        </Grid>
+
+        <Grid item xs={12}>
+          <Box className={classes.descriptionBox}>
+            <Typography>
+              <strong>
+                {
+                  <Trans
+                    defaults="You have {{amount}} newsletter statuses to confirm."
+                    values={{
+                      amount: loading ? '...' : totalCount,
+                    }}
+                  />
+                }
+              </strong>
+            </Typography>
+            <Typography>
+              {t(
+                'Contacts that appear here have an empty Newsletter Status and Partner Status set to Financial, Special, or Pray. Choose a newsletter status for contacts below.',
+              )}
+            </Typography>
+          </Box>
+        </Grid>
+        {!loading && data && !!numberOfContacts ? (
+          <>
+            <ButtonHeaderBox mb={0}>
+              <Box>
+                <Typography>
+                  <Trans
+                    defaults="<i>Showing <bold>{{numberOfContacts}}</bold> of <bold>{{totalCount}}</bold></i>"
+                    shouldUnescape
+                    values={{
+                      numberOfContacts,
+                      totalCount,
+                    }}
+                    components={{ bold: <strong />, i: <i /> }}
+                  />
+                </Typography>
+              </Box>
+              {(loading || updating) && <LoadingSpinner firstLoad={true} />}
+              <Box>
+                <Button
+                  variant="contained"
+                  onClick={() => null}
+                  disabled={updating || !numberOfContacts}
+                  sx={{ mr: 2 }}
+                >
+                  {
                     <Trans
-                      defaults="Showing <bold>{{value}}</bold> of <bold>{{value}}</bold>"
-                      shouldUnescape
+                      defaults="Confirm All ({{value}})"
                       values={{
                         value: numberOfContacts,
                       }}
-                      components={{ bold: <strong /> }}
                     />
-                  </Typography>
-                </Box>
-              </Grid>
-            </>
-          ) : (
-            <NoData tool="fixSendNewsletter" />
-          )}
-        </Grid>
-      ) : (
-        <CircularProgress style={{ marginTop: theme.spacing(3) }} />
-      )}
+                  }
+                </Button>
+              </Box>
+            </ButtonHeaderBox>
+            {contactsToFix?.map((contact) => (
+              <Contact
+                id={contact.id}
+                name={contact.name}
+                // need to fix this after changes to fix commitment info get merged
+                avatar={contact.avatar}
+                status={
```

Comment:
Since you are passing down 3 props for contact, you might as well pass down the whole contact object.